drivers/gpio/gpio_stm32.c

```diff
@@ -18,8 +18,6 @@
 #include <stm32_ll_system.h>
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/clock_control/stm32_clock_control.h>
-#include <pinmux/pinmux_stm32.h>
```

Comment:
We're actually using it but you can remove this one since it's also included in gpio_stm32.h.
But yes, once PINMUX deprecation period is complete, we'll need to move definitions from pinmux_stm32.h  to gpio_stm32.h